pandas/core/indexes/numeric.py

```diff
@@ -62,6 +62,13 @@ def _maybe_cast_slice_bound(self, label, side, kind):
         # we will try to coerce to integers
         return self._maybe_cast_indexer(label)
 
+    @Appender(_index_shared_docs['_shallow_copy'])
+    def _shallow_copy(self, values=None, **kwargs):
+        if values is not None and not self._can_hold_na:
```

Comment:
can you add a comment here on why we are doing this (yes its obvious to me, but not generally to future readers).

Reply:
(done)
ping